### ISSUE TITLE
Show file size and date modified in file explorer.

### ISSUE THREAD

**Issue by TheQuinbox** [microsoft/vscode#164033]
VSCode's file explorer currently doesn't show any attributes of files (at least not to a screen reader). I'm not able to confirm if it shows them visually, but I doubt it. Attributes such as file size and last modified date can be quite helpful when working on a code project. Is it possible to add these by chance? Thanks!

**Comment by gjsjohnmurray** [comment]
You are correct that these attributes aren't shown visually either. Previous request was #124115 in 2021.

**Comment by braillescreen** [comment]
This would be useful for me, too.

**Comment by TheQuinbox** [comment]
Hopefully this doesn't get closed, too. This would be an extremely useful feature.

**Comment by jamieoo** [comment]
Explorer can be set to sort by last modified date, but we can't actually see that date in explorer? 

The other sort orders (type + file) are immediately clear from their icons. I don’t see why date should be treated any different.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
:slightly_smiling_face: This feature request received a sufficient number of community upvotes and we moved it to our backlog. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by psulek** [comment]
hopefully vscode team will soon pick this request from backlog into real feature :-)

**Comment by lramos15** [comment]
There's a UX question here too. Most users don't care about this information. Is this something we show just on hover? Is this something where we configure what fields are shown. Right now the explorer is just a tree with singular entries as files, there's no "details"

**Comment by psulek** [comment]
For me , the best option is to align it to the right next to file/dir name, like this:
![image](https://user-images.githubusercontent.com/2369743/223521151-8ba78db3-5a35-44d0-97d8-0a68c37a87d8.png)

If if will be displayed always or on hover thats the config option i guess.

And is something like this possible now to do as extension via extension API ?

**Comment by Total-Reality** [comment]
> There's a UX question here too. Most users don't care about this information. Is this something we show just on hover? Is this something where we configure what fields are shown. Right now the explorer is just a tree with singular entries as files, there's no "details"

No, don't show this info on hover only. You could realize it like FileZilla - it is possible to choose which columns you need (file size, date modified and file type). Sorting would be usefull too... 
Put this settings in the explorer header please (beside the refresh button on my screen) or show this in the right-click dialogue on a folder.

![image](https://user-images.githubusercontent.com/5235894/223735982-f075bea4-64d9-46c3-b53f-27d9e945208f.png)

**Comment by jerryhall** [comment]
Has this feature been deployed yet? I'm new to VS but, not having the ability to see the owner, permissions, and date is puzzling.  I moved to VS from WinSCP and they have an excellent interface. Thanks.

**Comment by mukulpareek** [comment]
This would be such a useful feature, add columns such as last modified date and be able to click to sort by the modified date.  So most recent files one is working on will show up on top.  Just like in Jupyter lab - no different.  Right now it is such a pain to find a file.

**Comment by nranthony** [comment]
Why not pull some stock code to show what we see in Explorer: date, size, etc, right click to select columns, and sort on column click.  Standard stuff that's always nice to have.  I like to have sort by date modified so I know which files are more likely to have the recent/relevant notebooks and the such.  I wish I could have a perfect atomic commit workflow in place, but life's not that clean for me ;)
You could always 'level up' the standard Explorer options and have things like commit # (index back in time maybe); it would be nice to be able to see the commit info next to the files in some minimal way.

Look forward to this feature coming soon :)

**Comment by trendco** [comment]
Best like this:

![explorer infos](https://github.com/microsoft/vscode/assets/31956495/3fdac36c-b007-448e-bd48-57b6c4fac755)

with the Option to sort it up/down and with the possiblity to displayy/hide each column (filetype / date of last change / filesize / ...).

**Comment by sickerin** [comment]
![image](https://github.com/microsoft/vscode/assets/12108084/7dc5def1-ab4a-47b3-99c9-c9ada24ef288)
Jupyterlab has it. I've consistently found it helpful, especially on cloud servers and figuring out other user's projects, etc.

**Comment by yzliu01** [comment]
Also looking forward to the feature of showing file size and modified data or sorting by others.

**Comment by garikAsplund** [comment]
Yea, this is like super basic stuff and would immensely helpful to have, especially during optimization!! When is it coming out? Or is there an extension that does it already?

**Comment by dtap001** [comment]
ping! 
eta?
thanks!

**Comment by grimrog** [comment]
Hi, are there any news on this? 
I would also like to opt in to get the modified date next to the File name. 
Thanks.

**Comment by StephenAtKinetiq** [comment]
<img width="512" alt="Screenshot 2024-01-22 at 5 10 34 PM" src="https://github.com/microsoft/vscode/assets/117320428/a40e75ae-bb57-44c1-a22f-4649dad93a4e">
An excellent version of these features is available in an extension, Filesize Explorer.

**Comment by dynagence** [comment]
Dreamweaver is better, local and FTP 😄
Please add date and file size.
![image](https://github.com/microsoft/vscode/assets/3965049/2cd367be-3a88-4598-8ace-d76a0b4e7610)

**Comment by KeysAU** [comment]
Any ETA? is feature would make my life so much easier.

**Comment by ALtheHunter** [comment]
Hello,
I also like to see the file size and last modified in file explorer.

**Comment by fringemedia** [comment]
Where is this feature, it's such a basic necessity?!

**Comment by AdamPeterHiggins** [comment]
+1 for hoping this feature is added soon. It is crystal clear that the user base wants the option to display this information. Why the hold up?

**Comment by arturslab** [comment]
Maybe a good solution would be something like in PhpStorm? There is a file list with filenames and the grayed color of a file created time, size, and time from last access to the file (or something like that):
![image](https://github.com/microsoft/vscode/assets/40725161/97a60451-c1f5-4edf-9c53-54f877f43a58)

**Comment by dynagence** [comment]
And integrate a real FTP with this same informations. SFTP sucks and is full of bugs.
I don't understand how old software like Dreamweaver can be better than VSCcode. Unfortunately it doesn't support PHP8, otherwise I would never have switched.

**Comment by klingat** [comment]
+1 for this feature. I image it to be exactly like @psulek showed, as an optional setting.

**Comment by dynagence** [comment]
And be able to sort by name, size and date...

**Comment by NateFerrero** [comment]
+1 on this!

**Comment by shahzadir** [comment]
+1, extremely useful feature is missing

**Comment by hclever** [comment]
Up, Has this feature been developed yet?

**Comment by ecstrema** [comment]
See also https://github.com/microsoft/vscode/issues/216586

Personnally, i would show it on hover, but definetely not always, although I get why some people would want to.

**Comment by dynagence** [comment]
On hover is useless. You must display date and size and be able to sort. Also in FTP, like FileZilla or Dreamweaver. It's unusable, without it you have to open FileZilla.

**Comment by jmcgrory** [comment]
🗻 bump

**Comment by karkraeg** [comment]
+1 from me

**Comment by nicholaiTukanov** [comment]
+1 would be incredibly helpful

**Comment by bpisaacoff** [comment]
+1 from me

**Comment by psulek** [comment]
> See also https://github.com/microsoft/vscode/issues/216586
> 
> Personnally, i would show it on hover, but definetely not always, although I get why some people would want to.

This can be setting, to show on hover or statically

**Comment by FraserParlane** [comment]
+1!

**Comment by codegzx** [comment]
+1 for mr

**Comment by cpoptic** [comment]
Over. Three. Years.  That's how long it's been since this issue was mentioned in issue #124115, 

To implement a feature that just displays when a file was last modified and created.

Unbelievable that so much time has elapsed since such a basic feature has been requested.

I've been through several relationships in the time since this feature was first requested, the US economy has gone through an entire cycle in that time, and my clothes that were in style when this feature was first requested are no longer in style.

That's how long it's been.

 Why is it taking so long to implement what is essentially `ls -hal`?

**Comment by alexanderkryska** [comment]
I need this feature so much! I work in parallel with filezilla or dreamweaver, just because to see the file date. This is so stupid.

**Comment by dynagence** [comment]
Like everybody... VScode and SFTP suck...

**Comment by dynagence** [comment]
> I need this feature so much! I work in parallel with filezilla or dreamweaver, just because to see the file date. This is so stupid.

Like everybody... VScode and SFTP suck...

**Comment by FSGCorp** [comment]
Yes should be basic functionality of any editor. Should display date, time, size, etc. See attached screenshot from UltraEdit.
![Screenshot 2024-08-29 094816](https://github.com/user-attachments/assets/1cf8af33-a365-4afa-b49c-fe480e25f20b)

**Comment by BorgeJorge** [comment]
Actually it would be great to have basic Windows file explorer or Mac finder functionality: that is, show filename, date/time, file type, and maybe file size, and allow us to click on the column header to sort by any of those.

**Comment by dogfella** [comment]
+1 Git 'er done!

**Comment by r4dm** [comment]
+1

**Comment by richmd54** [comment]
+1

**Comment by dmnemch** [comment]
+1

**Comment by EonAndahalf** [comment]
+1

**Comment by oskaradermecker** [comment]
+1

**Comment by whiletrue57** [comment]
+1

**Comment by gianthk** [comment]
+1

**Comment by MCDengineering** [comment]
2 years since it was added to backlog...

**Comment by naourass** [comment]
I check if file size info was finally implemented at least 10 times a year.

**Comment by pranav-simplismart** [comment]
+1

**Comment by melonexe** [comment]
Hey all this is a great extension which allows for showing filesize in the explorer section. Did exactly what I wanted it to.

Confused me for a minute, but you have to look under the filesize explorer section in explorer

![Image](https://github.com/user-attachments/assets/156b59f5-c82d-427b-8467-81bd8b68d330)


https://marketplace.visualstudio.com/items?itemName=trietho.file-size-explorer

https://github.com/trietho/vscode-fliesize-explorer

**Comment by dynagence** [comment]
> Hey all this is a great extension which allows for showing filesize in the explorer section. Did exactly what I wanted it to.
> 
> Confused me for a minute, but you have to look under the filesize explorer section in explorer
> 
> ![Image](https://github.com/user-attachments/assets/156b59f5-c82d-427b-8467-81bd8b68d330)
> 
> https://marketplace.visualstudio.com/items?itemName=trietho.file-size-explorer
> 
> https://github.com/trietho/vscode-fliesize-explorer

It sucks ! You don't have dates, it's not in Explorer and you can't order... USELESS !

**Comment by melonexe** [comment]
> > Hey all this is a great extension which allows for showing filesize in the explorer section. Did exactly what I wanted it to.
> > Confused me for a minute, but you have to look under the filesize explorer section in explorer
> > ![Image](https://github.com/user-attachments/assets/156b59f5-c82d-427b-8467-81bd8b68d330)
> > https://marketplace.visualstudio.com/items?itemName=trietho.file-size-explorer
> > https://github.com/trietho/vscode-fliesize-explorer
> 
> It sucks ! You don't have dates, it's not in Explorer and you can't order... USELESS !

Works great for my Use case, you seem bitter, go for a walk xD

**Comment by doolim98** [comment]
+1

**Comment by Lobz** [comment]
+1

**Comment by ixqbar** [comment]
+1

**Comment by ixqbar** [comment]
<img width="301" alt="Image" src="https://github.com/user-attachments/assets/0eadd493-ed04-4c8b-9696-10d330604c55" />

**Comment by clarknova99** [comment]
+1

**Comment by waflessnet** [comment]
+1

**Comment by MendeBadra** [comment]
+1

**Comment by bticc** [comment]
+1

**Comment by mherman-amd** [comment]
+1

**Comment by calumk** [comment]
How does this not exist already??

**Comment by Margaretali** [comment]
+1

**Comment by tony-spartera** [comment]
Its been 3-4 years now - when can we expect to see file attributes in the explorer window (besides full path)? Thanks!

**Comment by gabrielbryk** [comment]
Why are +1 comments being downvoted? It just a way to indicate to devs that this is a feature many people want.

+1

**Comment by gjsjohnmurray** [comment]
+1 comments cause low-value notifications to everyone subscribed to the issue. The polite way to indicate you want a feature is to 👍 the issue description.

See https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions

**Comment by Total-Reality** [comment]
My original issue on this topic was submitted back on May 18, 2021:
https://github.com/microsoft/vscode/issues/124115

It's truly disappointing that it was closed – and that more than 4 years have now passed without any meaningful progress on this feature. This is a basic functionality that has been standard in many other editors for decades – and even in Windows Explorer, which is also a Microsoft product.

Given that VS Code is developed by Microsoft as well, it's hard to understand why such a fundamental feature – displaying and sorting file metadata (e.g. modification date) in the file explorer – is still missing.

We’re currently working on a design proposal to show how this feature could be integrated in a clear and useful way. We’ll share it here soon.

**Comment by shamseerahammedm** [comment]
+1

**Comment by NateFerrero** [comment]
Here is a POC showing just file sizes in the explorer

https://github.com/tagmein/vscode/pull/1

<img width="1200" height="800" alt="Image" src="https://github.com/user-attachments/assets/a4dc62a6-3d3e-486e-8e62-676b28e500d7" />

I am not very familiar with the project architecture so if anyone wants to run with this and contribute or a code review is appreciated

**Comment by Jy-EggRoll** [comment]
I have created an extension that meets your requirements. Now you can display file size and last modified date by hovering over them. For folders, it recursively calculates the size, number of contained folders, and number of contained files. However, it is worth noting that there is an issue with VS Code's file manager and the "modified" sort order; it often fails to refresh correctly. This extension is also affected by this problem. If you like, I will make a English version. Welcome!

Link: https://github.com/Jy-EggRoll/tree-enhancer

![](https://raw.githubusercontent.com/Jy-EggRoll/tree-enhancer/refs/heads/master/%E6%96%87%E4%BB%B6%E5%A4%B9%E6%95%88%E6%9E%9C%E6%BC%94%E7%A4%BA.png)

![](https://raw.githubusercontent.com/Jy-EggRoll/tree-enhancer/refs/heads/master/%E6%96%87%E4%BB%B6%E6%95%88%E6%9E%9C%E6%BC%94%E7%A4%BA.png)

![](https://github.com/Jy-EggRoll/tree-enhancer/blob/master/%E4%B8%8E%E5%85%B6%E4%BB%96%E6%82%AC%E6%B5%AE%E6%8F%90%E7%A4%BA%E5%8A%9F%E8%83%BD%E7%9A%84%E5%85%BC%E5%AE%B9%E6%80%A7.png)

**Comment by dynagence** [comment]
> I have created an extension that meets your requirements. Now you can display file size and last modified date by hovering over them. For folders, it recursively calculates the size, number of contained folders, and number of contained files. However, it is worth noting that there is an issue with VS Code's file manager and the "modified" sort order; it often fails to refresh correctly. This extension is also affected by this problem. If you like, I will make a English version. Welcome!
> 
> Link: https://github.com/Jy-EggRoll/tree-enhancer
> 
> ![](https://raw.githubusercontent.com/Jy-EggRoll/tree-enhancer/refs/heads/master/%E6%96%87%E4%BB%B6%E5%A4%B9%E6%95%88%E6%9E%9C%E6%BC%94%E7%A4%BA.png)
> 
> ![](https://raw.githubusercontent.com/Jy-EggRoll/tree-enhancer/refs/heads/master/%E6%96%87%E4%BB%B6%E6%95%88%E6%9E%9C%E6%BC%94%E7%A4%BA.png)
> 
> ![](https://github.com/Jy-EggRoll/tree-enhancer/blob/master/%E4%B8%8E%E5%85%B6%E4%BB%96%E6%82%AC%E6%B5%AE%E6%8F%90%E7%A4%BA%E5%8A%9F%E8%83%BD%E7%9A%84%E5%85%BC%E5%AE%B9%E6%80%A7.png)

No thanks, we need this information displayed permanently and above all to be able to sort it.

**Comment by Jy-EggRoll** [comment]
> No thanks, we need this information displayed permanently and, above all, to be able to sort it.

I am working on adding small upgrades to this extension. Now the extension supports almost any language, just by modifying the string template in the settings. The point you mentioned has also been troubling me for a long time, and I haven't made much substantial progress in implementing these features in the extension development. Unless I create my own file explorer view, I will strive to find various solutions.

**Comment by k-e-s-h-a-v** [comment]
hey @dynagence @NateFerrero @Total-Reality @trypsynth @gjsjohnmurray @braillescreen @jamieoo 
are you looking for something like this 

![Image](https://github.com/user-attachments/assets/96370570-b949-4552-871c-7837c7f43f0f)

here is an extension [Explorer++](https://marketplace.visualstudio.com/items?itemName=k-e-s-h-a-v.explorer-plus)

**Comment by BorgeJorge** [comment]
This is perfect @k-e-s-h-a-v, thank you!

**Comment by mukulpareek** [comment]
This is perfect! Thank you for this!

**Comment by Aaqu** [comment]
webstorm better

**Comment by astraetech** [comment]
Any updates on this issue? Switched to VSCode two months ago and a bit confused about the absence of this pretty obvious feature. Hope the devs get back to it soon
